scenariorun/src/main/kotlin/com/cosmotech/scenariorun/azure/ScenarioRunServiceImpl.kt

```diff
@@ -148,6 +151,20 @@ internal class ScenarioRunServiceImpl(
     }
   }
 
+  private fun updateScenarioLastRun(scenarioRun: ScenarioRun) {
+    var scenario =
+        scenarioApiService.findScenarioById(
+            scenarioRun.organizationId!!, scenarioRun.workspaceId!!, scenarioRun.scenarioId!!)
+
+    val lastRunId = scenario.lastRun!!.scenarioRunId
+    if (lastRunId == scenarioRun.id) {
```

Comment:
This rule is not clear, it is a guess rule on deletion. Moreover this updateScenarioLastRun function only handle this deletion case so it should be renamed deleteScenarioLastRunEvent- or something like this and remove this test.